modxcms/revolution
# Prevent OnResourceDelete from Incorrectly Firing on Every Resource Update

### What does it do?
Changed conditional logic in `checkDeletedStatus` method to correctly set `resourceDeleted` and `resourceUnDeleted` class properties only when the status of `deleted` has changed. The change casts both the incoming form and existing db value for `deleted` to boolean and removes the irrelevant and problematic test for null.

### Why is it needed?
Because strict comparison is being used and the form property `deleted` received from the Resource form will _never_ be null, the initial condition indicating a change in that property was always returning true.

### How to test
1. Create a plugin that logs a message for the `OnResourceDelete` and `OnResourceUndelete` events. Here's the plugin code I created for the test: 
```php
if ($modx->event->name == 'OnResourceDelete') {
    $modx->log(
        \modX::LOG_LEVEL_ERROR,
        "\r\tPLUGIN :: Resource delete event was fired for resource id ".$resource->get('id').'!'
    );
}
if ($modx->event->name == 'OnResourceUndelete') {
    $modx->log(
        \modX::LOG_LEVEL_ERROR,
        "\r\tPLUGIN :: Resource undelete event was fired for resource id ".$resource->get('id').'!'
    );
}
```
2. Choose an existing resource and change its `deleted` value (via the Resource editing form and the contextual menu in the tree) to verify the correct event fires (by examining your manager log).
3. Also change some other value in the Resource form _without_ changing the `deleted` value to verify that neither event fires.

### Related issue(s)/PR(s)
Resolves #16295.
